openssl/src/encrypt.rs

```diff
@@ -157,6 +158,32 @@ impl<'a> Encrypter<'a> {
         }
     }
 
+    /// Sets the RSA OAEP label.
+    ///
+    /// This is only useful for RSA keys.
+    ///
+    /// This corresponds to [`EVP_PKEY_CTX_set0_rsa_oaep_label`].
+    ///
+    /// [`EVP_PKEY_CTX_set0_rsa_oaep_label`]: https://www.openssl.org/docs/manmaster/man3/EVP_PKEY_CTX_set0_rsa_oaep_label.html
+    #[cfg(any(ossl102, libressl310))]
+    pub fn set_rsa_oaep_label(&mut self, label: CString) -> Result<(), ErrorStack> {
+        unsafe {
+            let label = label.into_bytes_with_nul().into_boxed_slice();
+            // Openssl takes ownership of the label and will free it if needed
+            let label = Box::leak(label);
```

Comment:
Allocations performed by Rust are not compatible with those performed by OpenSSL. This needs to call CRYPTO_malloc manually, like here: https://github.com/sfackler/rust-openssl/blob/9872a5dd7f476170372b9e119b8a6b681396b434/openssl/src/ssl/mod.rs#L3126-L3130.

Reply:
makes sense.